src/python/pants/reporting/zipkin_reporter.py

```diff
@@ -115,3 +120,26 @@ def end_workunit(self, workunit):
     if workunit in self._workunits_to_spans:
       span = self._workunits_to_spans.pop(workunit)
       span.stop()
+
+  def bulk_record_workunits(self, engine_workunits):
+    """A collection of workunits from v2 engine part"""
+    for workunit in engine_workunits:
+      duration = workunit['end_timestamp'] - workunit['start_timestamp']
+
+      span = zipkin_span(
+        service_name="pants",
+        span_name=workunit['name'],
+        duration=duration,
+        span_storage=self.span_storage,
+      )
+      span.zipkin_attrs = ZipkinAttrs(
+        trace_id=self.trace_id,
+        span_id=workunit['span_id'],
+        parent_span_id=workunit.get("parent_id", self.parent_id), # TODO change it when we properly pass parent id
```

Comment:
What is the effect of this? Does this mean that calls to the engine won't appear to have occurred under the workunit where someone called the v2 engine (ie, where they called `scheduler.product_request(..)`)? It would be good to include some more info in the comment if we're planning to land it this way.